single_tech_samples/databricks/sample4_ci_cd/README.md

```diff
@@ -1 +1,218 @@
+
+   - Active subscription with the following [resource providers](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/azure-services-resource-providers) enabled:
+     - Microsoft.Databricks
+3. [DevOps for Azure Databricks](https://marketplace.visualstudio.com/items?itemName=riserrad.azdo-databricks) It is the plugin of DevOps to support Azure Databricks deployments.
+
+### 2.2. Setup and deployment
+
+[Please check this document to the provision of databricks and other services](devops/README.md)
+
+### 2.3. Sample project Structure
+
+```txt
+
+├── devops
+|   ├── config
+|   ├── scripts
+|   ├── template
+|   |    └── jobs
+│   ├── README.md
+│   └── iac-create-environment-pipeline-arm.yml
+├── hybrid
+│   ├── README.md
+│   ├── devops
+│   │    ├──cd-pipeline.yml
+│   │    └──ci-pipeline.yml
+│   ├── common
+│   │    ├── __init__.py
+│   │    ├── module_a.py
+│   │    ├── module_b.py
+│   │    └── tests
+│   │          ├── module_a_test.py
+│   │          └── module_b_test.py
+│   ├── notebooks
+│   │    ├── main_notebook.py
+│   │    └── tests
+│   │          └── main_notebook_test.py
+│   └── setup.py
+├── notebook
+│   ├── README.md
+│   ├── devops
+│   │      ├──cd-pipeline.yml
+│   │      └──ci-pipeline.yml
+│   └── notebooks
+│          ├── main_notebook.py
+│          ├── main_notebook_sql.py
+│          ├── module_a_notebook.py
+│          ├── module_b_notebook.py
+│          └── tests
+│               ├── main_notebook_sql_test.py
+│               ├── main_notebook_test.py
+│               ├── module_a_notebook_test.py
+│               └── module_b_notebook_test.py
+├── python
+│   ├── README.md
+│   ├── devops
+│   │      ├──cd-pipeline.yml
+│   │      └──ci-pipeline.yml
+│   ├── common
+│   │    ├── __init__.py
+│   │    ├── module_a.py
+│   │    └── module_b.py
+│   ├── main.py
+│   ├── setup.py
+│   └── tests
+│        ├── integration
+│        │     └── main_test.py
+│        └── unit
+│              ├── module_a_test.py
+│              └── module_b_test.py
+├── pytest.ini
+├── README.md
+└── unit-requirements.txt
+
+```
+
+In this template we offer 5 different ways to build a Spark Application with Databricks notebooks or python files.
+In each sample code folder, there are
+
+- sample notebooks or python files
+- unit testing code
+- Azure DevOps pipeline yaml files
+
+#### How to use the template
+
+1. Here is a shell script to setup your project with the template
+
+```shell
+git clone https://github.com/Azure-Samples/modern-data-warehouse-dataops.git 
+cd modern-data-warehouse-dataops
+git checkout single-tech/databricks-ops
```

Comment:
We will need to update this prior to merge to master.

Reply:
Yes, it is. I will create a task to track it.